src/sparseml/pytorch/torchvision/train.py

```diff
@@ -533,6 +535,10 @@ def collate_fn(batch):
     # load params
     if checkpoint is not None:
         if "optimizer" in checkpoint and not args.test_only:
+            checkpoint["optimizer"] = _update_checkpoint_optimizer(
+                checkpoint_optim=checkpoint["optimizer"],
+                model=model,
+            )
             optimizer.load_state_dict(checkpoint["optimizer"])
```

Comment:
Thoughts on: if we detect transfer learning, don't load the optimizers state dict from the checkpoint? In theory since its a different dataset, would the momentums/etc from the state be invalid?

Reply:
Yeah after looking through our old image_classification code, it doesn't seem like optimizer ever loads its state_dict from a checkpoint.

Reply:
Sent in slack - yes I think we can  skip weight load in this case but a should at least warn